Unit.cpp

```diff
@@ -41,7 +42,7 @@ std::string Unit::ToString() const
 	std::string s = name + ": HP: " + std::to_string(hp) + ", DMG: " + std::to_string(dmg) + "\n";
 	return s;
 }
-Unit Unit::ParseUnit(const std::string &fileName)
+Unit *Unit::ParseUnit(const std::string &fileName)
```

Comment:
Az nem jelent érdemi változást, hogy a new máshova került.

Reply:
pointerekre úgy gondoljuk, hogy szükség van abból kifolyólag, hogy a dmg az egy const adattag a Unit-ban, tehát azonnal kell inicializálni. Viszont a try-ban el kell kapni a JsonParser által dobott exception-t is. Ha a try-ban hozok létre két Unit változót, akkor a scope miatt nem tudnának később harcolni.